mobile/src/screens/marketplace.js

```diff
@@ -564,9 +565,71 @@ class MarketplaceScreen extends Component {
     updateExchangeRate(this.state.fiatCurrency[1], 'DAI')
   }
 
-  onWebViewNavigationStateChange = state => {
+  _openDeepLinkUrlAttempt = async (
+    interceptUrlPredicate,
+    makeUrl,
+    timeControlVariableName
+  ) => {
+    // non interceptable url
+    if (!interceptUrlPredicate()) return
+
+    const url = makeUrl()
+    if (Linking.canOpenURL(url)) {
+      this.dappWebView.goBack()
+      // preventing multiple subsequent shares
+      if (
+        !this[timeControlVariableName] ||
+        new Date() - this[timeControlVariableName] > 3000
+      ) {
+        this[timeControlVariableName] = new Date()
+        return await Linking.openURL(url)
+      }
+    } else {
+      // can not open deep link url
+      return false
+    }
+  }
+
+  checkForShareNativeDialogInterception = async url => {
+    // natively tweet if possible on Android
+    await this._openDeepLinkUrlAttempt(
+      () =>
+        url.hostname === 'twitter.com' &&
+        url.pathname === '/intent/tweet' &&
+        Platform.OS === 'android',
+      () =>
+        `twitter://post?message=${encodeURIComponent(
```

Comment:
What happens with these kind of links if the user doesn't have Twitter or Facebook? Do they just fail silently?